service/history/historyEngine.go

```diff
@@ -1495,9 +1498,12 @@ Update_History_Loop:
 				}
 
 				if ai.StartedID == common.EmptyEventID {
-					// We haven't started the activity yet, we can cancel the activity right away.
+					// We haven't started the activity yet, we can cancel the activity right away and
+					// schedule a decision task to ensure the workflow makes progress.
 					msBuilder.AddActivityTaskCanceledEvent(ai.ScheduleID, ai.StartedID, *actCancelReqEvent.EventId,
 						[]byte(activityCancelationMsgActivityNotStarted), common.StringDefault(request.Identity))
+					isComplete = false
```

Comment:
I'm not sure overriding isComplete to false is the right thing to do over here.  If a decider makes request_cancel and complete decision in a single transaction we would still want to honor completion.